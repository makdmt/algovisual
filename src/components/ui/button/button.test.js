import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button'
import renderer from 'react-test-renderer'

describe('Тесты компонента Button', () => {
    it('Рендеринг Button без надписи соответствует снепшоту', () => {
        const tree = renderer
            .create(<Button />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Button с надписью соответствует снепшоту', () => {
        const tree = renderer
            .create(<Button text="Запуск" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Button с аттрибутом disabled cоответствует снепшоту', () => {
        const tree = renderer
            .create(<Button disabled={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Button с индикацией загрузки cоответствует снепшоту', () => {
        const tree = renderer
            .create(<Button isLoader={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('По клику на Button запускается колбек, переданный ему в пропс OnClick', () => {

        const callBack = () => {
            alert('Это колбек');
        }

        window.alert = jest.fn();
        render(<Button text="Запуск" onClick={callBack} />)
        const btn = screen.getByText('Запуск');
        fireEvent.click(btn);
        expect(window.alert).toHaveBeenCalledWith('Это колбек');
    });
})

