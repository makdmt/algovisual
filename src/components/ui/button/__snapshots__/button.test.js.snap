// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Button Рендеринг Button без надписи соответствует снепшоту 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тесты компонента Button Рендеринг Button с аттрибутом disabled cоответствует снепшоту 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тесты компонента Button Рендеринг Button с индикацией загрузки cоответствует снепшоту 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тесты компонента Button Рендеринг Button с надписью соответствует снепшоту 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Запуск
  </p>
</button>
`;
